mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,256 @@ def get_estimation_distribution(
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray, NDArray, RegressorMixin]
+        Well initiated and typed :
+            - X
+            - y
+            - y_pred
+            - residual_estimator
+            - random_state
+        """
+        residual_estimator = self._check_estimator(
+            self.residual_estimator
+        )
+        random_state = check_random_state(self.random_state)
+        X, y, y_pred = indexable(X, y, y_pred)
+        X = np.array(X)
+        y = np.array(y)
+        y_pred = np.array(y_pred)
+        return X, y, y_pred, residual_estimator, random_state
+
+    def _fit_residual_estimator(
+        self,
+        residual_estimator_: RegressorMixin,
+        X: NDArray,
+        y: NDArray,
+        y_pred: NDArray,
+    ) -> Tuple[NDArray, NDArray]:
+        """
+        Fit the residual estimator and returns the indexes used for the
+        training of the base estimator and those needed for the calibration.
+
+        Parameters
+        ----------
+        X: NDArray
+            All the observed values used in the general fit.
+
+        y: NDArray
+            All the observed targets used in the general fit.
+
+        y_pred: NDArray
+            Predicted targets.
+
+        Returns
+        -------
+        RegressorMixin
+            Fitted residual estimator
+        """
+        residuals = np.abs(np.subtract(y, y_pred))
+        targets = np.log(np.maximum(
+            residuals,
+            np.full(residuals.shape, self.eps)
+        ))
+
+        residual_estimator_ = residual_estimator_.fit(X, targets)
+
+        return residual_estimator_
+
+    def get_signed_conformity_scores(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        y_pred: ArrayLike
+    ) -> NDArray:
+        """
+        Computes the signed conformity score = (y - y_pred) / r_pred.
+        r_pred being the predicted residual (y - y_pred) of the estimator.
+        It is calculated by a model (``residual_estimator_``) that learns
+        to predict this residual.
+
+        The learning is done with the log of the residual and later we
+        use the exponential of the prediction to avoid negative values.
+        """
+        (X, y, y_pred,
+         self.residual_estimator_,
+         random_state) = self._check_parameters(X, y, y_pred)
+
+        full_indexes = np.argwhere(
+            np.logical_not(np.isnan(y_pred))
+        ).reshape((-1,))
+
+        if not self.prefit:
+            cal_indexes, res_indexes = train_test_split(
+                full_indexes,
+                test_size=self.split_size,
+                random_state=random_state,
+            )
+            self.residual_estimator_ = self._fit_residual_estimator(
+                clone(self.residual_estimator_),
+                X[res_indexes], y[res_indexes], y_pred[res_indexes]
+            )
+        else:
+            cal_indexes = full_indexes
+
+        residuals_pred = np.maximum(
```

Comment:
In the case where the user give a pre-fitted residual estimator, he do not necessarily fit his model using the log of the true residual, hence this exponential might cause some problems. So either we specify very clearly that the model should be fitted with the logarithm so that all users get the information directly, either we move this exponential elsewhere